### PR TITLE
Ivanlopez feature/prevent weak password auth

### PR DESCRIPTION
**Pull request by tlovett1**
Replaces #34

### CODE REVIEW

**Comment by ivanlopez** [comment]
@tlovett1 @helen I've updated the admin notice description with Melissa's recommendation.

File: includes/authentication.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * @param string  $password Password
+ *
+ * @return \WP_User|\WP_Error
+ */
+function prevent_weak_password_auth( $user, $username, $password ) {
+	$test_tlds = array( 'test', 'dev', 'local', '' );
+	$tld       = preg_replace( '#^.*\.(.*)$#', '$1', wp_parse_url( site_url(), PHP_URL_HOST ) );
+
+	if ( ! in_array( $tld, $test_tlds, true ) && in_array( strtolower( trim( $password ) ), weak_passwords(), true ) ) {
+		return new \WP_Error( 'Auth Error', sprintf( '%s <a href="%s">%s</a> %s',
```
Comment (helen) [inline includes/authentication.php]: It probably doesn't matter so much for this plugin practically-speaking but strings can't be internationalized piece-wise like this. Only the URL should be a placeholder. I also see that @jakemgold left a comment to the same affect as I was writing this one, but I don't think "recently improved security measures" is correct either.

**Comment by jakemgold** [comment]
I'm ok to ship, but I'm not sure I love the message: "in order to meet recently improved security measures" - this plugin is going to be used for years. Unless we have a flag to come back to this, this is going to seem off, and at worst, confusing ("did I miss a policy change?"). Can we just say "in order to meet current security measures" ?

File: includes/authentication.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+add_filter( 'authenticate', __NAMESPACE__ . '\prevent_weak_password_auth', 30, 3 );
+
+/**
+ * List of the most popular weak passwords
+ *
+ * @return array
+ */
+function weak_passwords() {
```
Comment (helen) [inline includes/authentication.php]: I'm not really excited about having a list of (bad) passwords in here like this and including profanity in something deployed to client sites. Something like zxcvbn might be overkill on the other end of the spectrum, but maybe this should at least be hashed?

Comment (helen) [reply includes/authentication.php]: @ivanlopez Could you let me know where you got this list from?

Comment (ivanlopez) [reply includes/authentication.php]: @helen it was from this article about the most popular passwords. http://fortune.com/2017/12/19/the-25-most-used-hackable-passwords-2017-star-wars-freedom/

**Comment by tlovett1** [comment]
@jakemgold changes made.

**Pull request merged**